frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretForm.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikValues, useFormikContext } from 'formik';
+import { ActionGroup, Button, ButtonVariant, TextInputTypes } from '@patternfly/react-core';
+import { CheckIcon, CloseIcon } from '@patternfly/react-icons';
+import { ButtonBar } from '@console/internal/components/utils';
+import {
+  SecretType,
+  BasicAuthSubform,
+  SSHAuthSubform,
+  CreateConfigSubform,
+} from '@console/internal/components/secrets/create-secret';
+import { DropdownField, InputField } from '@console/shared';
+import SecretAnnotation from './SecretAnnotation';
+import './SecretForm.scss';
+
+const authTypes = {
+  [SecretType.dockerconfigjson]: 'Image Registry Credentials',
+  [SecretType.basicAuth]: 'Basic Authentication',
+  [SecretType.sshAuth]: 'SSH Key',
+};
+
+const renderSecretForm = (
+  type: string,
+  stringData: {
+    [key: string]: any;
+  },
+  onDataChanged: (event: Event) => void,
+) => {
+  switch (type) {
+    case SecretType.basicAuth:
+      return (
+        <BasicAuthSubform onChange={onDataChanged} stringData={stringData[SecretType.basicAuth]} />
+      );
+    case SecretType.sshAuth:
+      return (
+        <SSHAuthSubform onChange={onDataChanged} stringData={stringData[SecretType.sshAuth]} />
+      );
+    case SecretType.dockerconfigjson:
+      return (
+        <CreateConfigSubform
+          onChange={onDataChanged}
+          stringData={stringData[SecretType.dockerconfigjson]}
+        />
+      );
+    default:
+      return null;
+  }
+};
+
+const SecretForm: React.FC<FormikValues> = ({
+  handleSubmit,
+  handleReset,
+  status,
+  isSubmitting,
+}) => {
+  const { values, setFieldValue } = useFormikContext<FormikValues>();
+  const [stringData, setStringData] = React.useState({
+    [SecretType.basicAuth]: {},
+    [SecretType.sshAuth]: {},
+    [SecretType.dockerconfigjson]: {},
+  });
+
+  const setValues = (type) => {
+    type === SecretType.dockerconfigjson
+      ? setFieldValue(
+          'formData',
+          _.mapValues({ '.dockerconfigjson': stringData[type] }, JSON.stringify),
+        )
+      : setFieldValue('formData', stringData[type]);
+  };
+
+  const onDataChanged = (event) => {
```

Comment:
Not an event... it's just the string value.

```suggestion
  const onDataChanged = (value: string) => {
```